pkgs/stdenv/generic/setup.sh

```diff
@@ -806,20 +806,17 @@ _defaultUnpack() {
     local fn="$1"
 
     if [ -d "$fn" ]; then
-
         # We can't preserve hardlinks because they may have been
         # introduced by store optimization, which might break things
         # in the build.
         cp -pr --reflink=auto -- "$fn" "$(stripHash "$fn")"
-
     else
-
         case "$fn" in
             *.tar.xz | *.tar.lzma | *.txz)
                 # Don't rely on tar knowing about .xz.
                 xz -d < "$fn" | tar xf -
                 ;;
-            *.tar | *.tar.* | *.tgz | *.tbz2 | *.tbz)
+            *.tar | *.tar.* | *.tgz | *.tbz2 | *.tbz | *.bz2)
```

Comment:
Other lines are just stylistically updating for consistency while we're here rebuilding the world; most other functions in stdenv's setup.sh don't have padding whitespace.  

A local run for me of `tar xf out.bz2` works just fine for me on NixOS, using `tar v1.32`.  Not sure if there's a reason we're excluding bz2?  It seems to be used for distribution of a bunch of the bootstrap binaries, alongside `xz`:
https://nixos.org/releases/nix/nix-2.3.2/